### PR TITLE
Protocol 3.0: ContainerType enum and registries

### PR DESCRIPTION
**Pull request by Konicai**

### CODE REVIEW

File: bootstrap/spigot/src/main/java/org/geysermc/geyser/platform/spigot/GeyserSpigotPlugin.java [hunk 1]
```diff
@@ -42,6 +42,7 @@
 import org.bukkit.permissions.PermissionDefault;
 import org.bukkit.plugin.Plugin;
 import org.bukkit.plugin.java.JavaPlugin;
+import org.cloudburstmc.math.MathImplementationLoader;
```
Comment (Camotoy) [inline bootstrap/spigot/src/main/java/org/geysermc/geyser/platform/spigot/GeyserSpigotPlugin.java:45]: Leftover import.

File: core/src/main/java/org/geysermc/geyser/registry/type/BlockMappings.java [hunk 2]
```diff
@@ -75,6 +76,10 @@ public BlockDefinition getItemFrame(NbtMap tag) {
     }
 
     public boolean isItemFrame(BlockDefinition definition) {
-        return this.itemFrames.containsKey(definition.getRuntimeId());
+        if (definition instanceof SimpleBlockDefinition def) {
```
Comment (Camotoy) [inline core/src/main/java/org/geysermc/geyser/registry/type/BlockMappings.java:79]: Not sure if I like this. Will merge without it being changed but BlockDefinition should always be a certain type.

Comment (Konicai) [reply core/src/main/java/org/geysermc/geyser/registry/type/BlockMappings.java:79]: > BlockDefinition should always be a certain type.

In regards to the packet translator that uses this?

Comment (Camotoy) [reply core/src/main/java/org/geysermc/geyser/registry/type/BlockMappings.java:79]: Checking for instanceof here implies that BlockDefinition might not be SimpleBlockDefinition, when (if I am correct) it'll always be SimpleBlockDefinition within Geyser.

Comment (Konicai) [reply core/src/main/java/org/geysermc/geyser/registry/type/BlockMappings.java:79]: Ah - I guess we could add a warning if it isn't?

**Pull request merged**